src/main.rs

```diff
@@ -58,43 +53,42 @@ const LINKNAME_LEN: usize = 100;
 const MAGIC_LEN: usize = 6;
 
 impl Header {
-    fn from_block(block: [u8;512]) -> Header {
+    fn from_block(block: [u8; 512]) -> Header {
         Header { block }
     }
 
-    fn has_magic(self: &Header) -> bool {
+    fn has_magic(&self) -> bool {
         let maybe_magic = String::from_utf8(self.magic_field().to_vec());
 
         match maybe_magic {
-            Ok(actual) => actual == String::from("ustar\0"),
+            Ok(actual) => actual == "ustar\0",
             _ => false,
         }
     }
 
-    fn is_null(self: &Header) -> bool {
-        let offset = NAME_LEN + MODE_LEN + UID_LEN + GID_LEN + SIZE_LEN +
```

Comment:
I wish rustfmt had left this alone. I prefer an 80 column width for languages like Rust that don't have a lot of nesting. It lets me fit two files and the dock side by side on my laptop.